packages/components/Callout/src/Callout.types.ts

```diff
@@ -92,6 +101,18 @@ export interface ICalloutProps extends ICalloutTokens {
    */
   onDismiss?: () => void;
 
+  /**
+   * Callback invoked during callout dismissal; if set, focus will not be restored by the callout and onRestoreFocus must
+   * result in focus being moved to the appropriate focusable target.
+   *
+   * The callee should carefully consider their scenarios to avoid dropping focus, or inappropriately
+   * moving focus from another component.  Focus is not guaranteed to have entered the React-Native surface at all, and
+   * this callback is most appropriate for components strictly controlling focus.
+   *
+   * restoreFocusEvent.nativeEvent.containsFocus is true if the Callout had focus while being dismissed.
+   */
+  onRestoreFocus?: (restoreFocusEvent: OnRestoreFocusEvent) => void;
```

Comment:
I'm not sure I actually understand the purpose here (on first glance seems excessive) - having talked offline it sounded like this was to match a FluentUI (web) API, right? When would the corresponding native event get dispatched (I'm wondering whether it would be after callout teardown -- potentially too late and getting you in weird timing situations)? Is `onDismiss` too early/late for consumers to do any imperative focusing of their own + tracking focused state should be relatively easy for wrapping components.

Reply:
so this replaces the deprecated prop doNotRestoreFocus : boolean , but makes the same native-side adjustment of not attempting to place focus back where it was (note: focus will still end up 'somewhere' but the behavior is different).  When intentionally managing focus for complex situations, this is useful, and yeah I think it's a bit redundant w/ onDismiss but onDismiss w/ 'containsFocus' info is a little bit "why is this a thing", so two callbacks are relatively fine on top of adding the native-side behavior.

Reply:
Yeah I suppose it's just a matter of how much tracking you want to offload to the wrapping component. Still seems a little weird to me, but I will not get in the way of API alignment

Reply:
it's more enabling the offloading; this is a niche-use API, the default should be appropriate for the vast majority of uses